lambda-natchez/src/main/scala/feral/lambda/natchez/TracedLambda.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2021 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package feral.lambda.natchez
+
+import cats.effect.kernel.MonadCancelThrow
+import cats.syntax.all._
+import feral.lambda.Lambda
+import natchez.EntryPoint
+import natchez.Span
+
+object TracedLambda {
+
+  def apply[F[_]: MonadCancelThrow, Event: KernelSource, Result](entryPoint: EntryPoint[F])(
+      lambda: Span[F] => Lambda[F, Event, Result]): Lambda[F, Event, Result] = {
```

Comment:
Given a parametric `Lambda`, wouldn't the better type here be something like `Lambda[Kleisli[F, Span[F], *], Event, Result]`? Could also bring in Cats MTL and do an `Ask[F, Span[F]]`, though that starts to get odd.

Reply:
Not necessarily. natchez also supports tracing with `IOLocal` instead of Kleisli which IIUC is also more performant. I played with a few other ideas in https://github.com/typelevel/feral/pull/50. Also hoping that https://github.com/tpolecat/natchez/pull/448 becomes a thing.

Reply:
Oh btw, I now have an MTL lambda concept in https://github.com/typelevel/feral/pull/60. Actually this is my favorite so far, and I think http4s will be moving in this direction as well.